synapseutils/sync.py

```diff
@@ -671,58 +660,53 @@ async def _upload_item_async(
         store_args,
         finished_items,
     ) -> File:
-        try:
-            await self._file_semaphore_async.acquire()
-            used_activity = []
-            executed_activity = []
-            for used_item in used + executed:
-                possible_file = finished_items.get(used_item, None)
-                if possible_file:
-                    used_item = possible_file.id
-                if is_url(used_item):
-                    if used_item in used:
-                        used_activity.append(UsedURL(url=used_item))
-                    else:
-                        executed_activity.append(UsedURL(url=used_item))
-
-                # -- Synapse Entity ID (assuming the string is an ID)
-                elif isinstance(used_item, str):
-                    if not is_synapse_id_str(used_item):
-                        raise ValueError(f"{used_item} is not a valid Synapse id")
-                    synid, version = get_synid_and_version(
-                        used_item
-                    )  # Handle synapseIds of from syn234.4
-                    target_version = None
-                    if version:
-                        target_version = int(version)
-                    if used_item in used:
-                        used_activity.append(
-                            UsedEntity(
-                                target_id=synid, target_version_number=target_version
-                            )
-                        )
-                    else:
-                        executed_activity.append(
-                            UsedEntity(
-                                target_id=synid, target_version_number=target_version
-                            )
+        used_activity = []
+        executed_activity = []
+        for used_item in used + executed:
+            possible_file = finished_items.get(used_item, None)
+            if possible_file:
+                used_item = possible_file.id
+            if is_url(used_item):
+                if used_item in used:
+                    used_activity.append(UsedURL(url=used_item))
+                else:
+                    executed_activity.append(UsedURL(url=used_item))
+
+            # -- Synapse Entity ID (assuming the string is an ID)
+            elif isinstance(used_item, str):
+                if not is_synapse_id_str(used_item):
+                    raise ValueError(f"{used_item} is not a valid Synapse id")
+                synid, version = get_synid_and_version(
+                    used_item
+                )  # Handle synapseIds of from syn234.4
+                target_version = None
+                if version:
+                    target_version = int(version)
+                if used_item in used:
+                    used_activity.append(
+                        UsedEntity(
+                            target_id=synid, target_version_number=target_version
                         )
+                    )
                 else:
```

Comment:
might it be safer to make this an elif? i.e. `elif used_item in executed:`

Reply:
Great point, I will be planning to clean all of this code up in the next PR. I wanted to keep this focused on the semaphore change.

Reply:
Oh, excellent, thanks for limiting the PR scope! Feel free to add me as a reviewer for the other one.